flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/SlotStatus.java

```diff
@@ -46,13 +47,21 @@
 	/** if the slot is allocated, jobId identify which job this slot is allocated to; else, jobId is null */
 	private final JobID jobID;
 
-	public SlotStatus(SlotID slotID, ResourceProfile profiler) {
-		this(slotID, profiler, null, null);
+	/** Gateway to the TaskManager which reported the SlotStatus */
+	private final TaskExecutorGateway taskExecutorGateway;
```

Comment:
The `SlotStatus` is no longer serializable with this field. Where does the `SlotStatus` come from? If it's coming from the `TaskExecutor`, then the `taskExecutorGateway` has to be retrieved on the `ResourceManager` side.